### ISSUE TITLE
Error when using FP16 or Mixed precision

### ISSUE THREAD

**Issue by Vaccummer** [state-spaces/mamba#474]
When I train the model of dtype fp16 or mixed precision on  V100 sxm2 16G, loss.backward() return error:

python: /project/lib/Analysis/Allocation.cpp:40: std::pair<llvm::SmallVector<unsigned int>, llvm::SmallVector<unsigned int> > mlir::triton::getCvtOrder(mlir::Attribute, mlir::Attribute): Assertion `!(srcMmaLayout && dstMmaLayout) && "Unexpected mma -> mma layout conversion"' failed.

**Comment by tridao** [comment]
Triton doens't support V100 very well

**Comment by Vaccummer** [comment]
> Triton doens't support V100 very well

Thanks for reply, and is there any solution or alternative of Triton?

**Comment by tridao** [comment]
There's a reference implementation in pytorch but would probably be quite a bit slower

**Comment by sascha-kirch** [comment]
Hi @tridao, just to confirm and for documentation, we had the same error on an older Laptop GPU and changing back to full-precission resolved the issue for us.

Cheers,
Sascha
